src/api/entities/SecurityToken/CorporateActions/index.ts

```diff
@@ -15,9 +24,31 @@ export class CorporateActions extends Namespace<SecurityToken> {
   constructor(parent: SecurityToken, context: Context) {
     super(parent, context);
 
+    const { ticker } = parent;
+
     this.distributions = new Distributions(parent, context);
+
+    this.modifyCorporateActionAgent = createProcedureMethod(
+      args => [modifyCorporateActionAgent, { ticker, ...args }],
+      context
+    );
   }
 
+  /**
+   * Assign a new corporate action agent for the Security Token
+   *
+   * @param args.target - identity to be set as corporate action agent
+   * @param args.requestExpiry - date at which the authorization request to modify the corporate action agent expires (optional, never expires if a date is not provided)
+   *
+   * @note this may create AuthorizationRequest which have to be accepted by
+   *   the corresponding Account. An Account or Identity can
+   *   fetch its pending Authorization Requests by calling `authorizations.getReceived`
+   *
+   * @note required role:
+   *   - Security Token Owner
+   */
```

Comment:
```suggestion
  /**
   * Assign a new Corporate Actions Agent for the Security Token
   *
   * @param args.target - identity to be set as Corporate Actions Agent
   * @param args.requestExpiry - date at which the authorization request to modify the Corporate Actions Agent expires (optional, never expires if a date is not provided)
   *
   * @note this may create AuthorizationRequests which have to be accepted by
   *   the corresponding Account. An Account or Identity can
   *   fetch its pending Authorization Requests by calling `authorizations.getReceived`
   *
   * @note required role:
   *   - Security Token Owner
   */
```